engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetInputTextureFromFBO.java

```diff
@@ -175,55 +168,14 @@ public boolean equals(Object other) {
     }
 
     /**
-     * Instances of this class do the actual work of activating a given texture unit,
-     * binding the appropriate texture to it and configuring a material to use it as input.
+     * Activates the texture unit GL_TEXTURE0 + textureSlot, binds the GL_TEXTURE_2D identified by textureId to it
+     * and sets the material provided on construction to sample the texture via the parameterName also provided on
+     * construction.
      */
-    protected class Task implements RenderPipelineTask {
-        private final int textureSlot;
-        private final Material material;
-        private final String shaderParameterName;
-        private int textureId;
-
-        /**
-         * Constructs an instance of this inner class, according to the given parameters.
-         *
-         * @param textureSlot an integer indirectly identifying a texture unit on the GPU (textureUnit = GL_TEXTURE0 + textureSlot).
-         * @param textureId the opengl id of a texture, usually obtained via glGenTextures().
-         * @param materialUrn an URN identifying a material.
-         * @param shaderParameterName the name of the variable in the shader program used to sample the texture.
-         */
-        private Task(int textureSlot, int textureId, ResourceUrn materialUrn, String shaderParameterName) {
-            this.textureSlot = textureSlot;
-            this.textureId = textureId;
-            this.material = getMaterial(materialUrn);
-            this.shaderParameterName = shaderParameterName;
-        }
-
-        /**
-         * This method is used when FBOs have changed and the id of the texture attached to one needs refreshing.
-         *
-         * @param textureId an integer identifying a texture buffer on the GPU, originally obtained via glGenTextures().
-         */
-        protected void setTextureId(int textureId) {
-            this.textureId = textureId;
-        }
-
-        /**
-         * Activates the texture unit GL_TEXTURE0 + textureSlot, binds the GL_TEXTURE_2D identified by textureId to it
-         * and sets the material provided on construction to sample the texture via the parameterName also provided on
-         * construction.
-         */
-        @Override
-        public void execute() {
-            glActiveTexture(GL_TEXTURE0 + textureSlot);
-            glBindTexture(GL_TEXTURE_2D, textureId);
-            material.setInt(shaderParameterName, textureSlot, true);
-        }
-
-        @Override
-        public String toString() {
-            return String.format("%30s: slot %s, texture %s, material %s, parameter %s", this.getClass().getSimpleName(),
-                    textureSlot, textureId, material.getUrn().toString(), shaderParameterName);
-        }
+    @Override
+    public void process() {
+        glActiveTexture(GL_TEXTURE0 + textureSlot);
+        glBindTexture(GL_TEXTURE_2D, textureId);
+        material.setInt(parameterName, textureSlot, true);
```

Comment:
Similar to a previous comment (but not yet to be tackled), this is a class that mixes opengl state changes with material state changes. Perhaps we can add a TODO in both cases? I.e.:

// TODO: split this class into two - one for opengl's global state change and one for the specific material state change.